src/data/finset/basic.lean

```diff
@@ -2085,6 +2125,9 @@ end
 
 @[simp] theorem card_attach {s : finset α} : card (attach s) = card s := multiset.card_attach
 
+@[simp] lemma attach_nonempty_iff (s : finset α) : s.attach.nonempty ↔ s.nonempty :=
+by simp [← card_pos]
```

Comment:
You can also prove this more directly using `by simp [finset.nonempty]`, which lets you move this lemma next to the other lemmas about attach rather than near the lemmas about card, since this isn't about cardinalities. You should also add

@[simp] lemma attach_eq_empty_iff (s : finset α) :
  s.attach = ∅ ↔ s = ∅ :=
by simpa [eq_empty_iff_forall_not_mem]